From b61e548125cae41a3d4203b4d2928856f425e002 Mon Sep 17 00:00:00 2001
From: Lucas Alvares Gomes <lucasagomes@gmail.com>
Date: Thu, 4 Jul 2013 11:17:29 +0100
Subject: [PATCH 1/1] Restart rabbitmq-server in case of failure

Tells systemd to restart the rabbitmq-server in case of failure

Change-Id: If5416ecc636e199d639be2c6bb8777326e7ea659
---
 elements/rabbitmq-server/install.d/20-rabbitmq-server | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/elements/rabbitmq-server/install.d/20-rabbitmq-server b/elements/rabbitmq-server/install.d/20-rabbitmq-server
index 2001a08..ab4e56a 100755
--- a/elements/rabbitmq-server/install.d/20-rabbitmq-server
+++ b/elements/rabbitmq-server/install.d/20-rabbitmq-server
@@ -23,6 +23,9 @@ if [ $DISTRO = "Fedora" ]; then
     # Delay the rc-local.service start-up until rabbitmq-server.service is started up
     sed -i 's/\[Unit\]/\[Unit\]\nBefore=rc-local.service/g' /lib/systemd/system/rabbitmq-server.service
 
+    # Respawn rabbitmq-server in case the process exits with an nonzero exit code
+    sed -i 's/\[Service\]/\[Service\]\nRestart=on-failure/g' /lib/systemd/system/rabbitmq-server.service
+
     # Make systemd take notice of it
     systemctl daemon-reload
 
-- 
1.8.1.4

